File: search_history.js
const fs = require('fs');
const path = require('path');

const historyDir = 'C:/Users/user/AppData/Roaming/Code/User/History';
if (!fs.existsSync(historyDir)) {
    console.error('History dir not found');
    process.exit(1);
}

let targetDirs = [];
const dirs = fs.readdirSync(historyDir);

for (const dir of dirs) {
    const dirPath = path.join(historyDir, dir);
    if (!fs.statSync(dirPath).isDirectory()) continue;
    const entriesPath = path.join(dirPath, 'entries.json');
    if (fs.existsSync(entriesPath)) {
        try {
            const entriesData = JSON.parse(fs.readFileSync(entriesPath, 'utf-8'));
            if (entriesData.resource && entriesData.resource.endsWith('index.html')) {
                if (entriesData.entries) {
                    for (const entry of entriesData.entries) {
                        const filePath = path.join(dirPath, entry.id);
                        if (fs.existsSync(filePath)) {
                            targetDirs.push({ filePath, timestamp: entry.timestamp });
                        }
                    }
                }
            }
        } catch (e) {
            // ignore
        }
    }
}

targetDirs.sort((a, b) => b.timestamp - a.timestamp);

let foundPath = null;
for (const entry of targetDirs) {
    const content = fs.readFileSync(entry.filePath, 'utf-8');
    if (content.includes('홍길동') || content.toLowerCase().includes('welcome')) {
        foundPath = entry.filePath;
        break;
    }
}

if (foundPath) {
    console.log('FOUND:', foundPath);
    console.log('CONTENT_START');
    console.log(fs.readFileSync(foundPath, 'utf-8'));
    console.log('CONTENT_END');
} else {
    console.log('Not found');
}

File: search_history.py
import os
import json

appdata = os.environ.get('APPDATA')
if appdata:
    history_dir = os.path.join(appdata, 'Code', 'User', 'History')
else:
    history_dir = os.path.expanduser('~/.config/Code/User/History')

target = None

if os.path.exists(history_dir):
    for d in os.listdir(history_dir):
        p = os.path.join(history_dir, d)
        if os.path.isdir(p):
            e_path = os.path.join(p, 'entries.json')
            if os.path.exists(e_path):
                try:
                    with open(e_path, 'r', encoding='utf-8') as f:
                        data = json.load(f)
                    if data.get('resource', '').endswith('index.html'):
                        for entry in data.get('entries', []):
                            entry_path = os.path.join(p, entry.get('id', ''))
                            if os.path.exists(entry_path):
                                with open(entry_path, 'r', encoding='utf-8', errors='ignore') as ef:
                                    content = ef.read()
                                if '홍길동' in content or 'welcome' in content.lower():
                                    target = content
                                    break
                except Exception as e:
                    pass
            if target:
                break

with open('result.txt', 'w', encoding='utf-8') as f:
    if target:
        f.write(target)
    else:
        f.write('NOT FOUND')
